x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/okta.go

```diff
@@ -0,0 +1,334 @@
+type HAL map[string]any
+
+// Profile is an Okta user's profile.
+//
+// See https://developer.okta.com/docs/reference/api/users/#profile-object for details.
+type Profile struct {
+	Login             string  `json:"login"`
+	Email             string  `json:"email"`
+	SecondEmail       *string `json:"secondEmail,omitempty"`
+	FirstName         *string `json:"firstName,omitempty"`
+	LastName          *string `json:"lastName,omitempty"`
+	MiddleName        *string `json:"middleName,omitempty"`
+	HonorificPrefix   *string `json:"honorificPrefix,omitempty"`
+	HonorificSuffix   *string `json:"honorificSuffix,omitempty"`
+	Title             *string `json:"title,omitempty"`
+	DisplayName       *string `json:"displayName,omitempty"`
+	NickName          *string `json:"nickName,omitempty"`
+	ProfileUrl        *string `json:"profileUrl,omitempty"`
+	PrimaryPhone      *string `json:"primaryPhone,omitempty"`
+	MobilePhone       *string `json:"mobilePhone,omitempty"`
+	StreetAddress     *string `json:"streetAddress,omitempty"`
+	City              *string `json:"city,omitempty"`
+	State             *string `json:"state,omitempty"`
+	ZipCode           *string `json:"zipCode,omitempty"`
+	CountryCode       *string `json:"countryCode,omitempty"`
+	PostalAddress     *string `json:"postalAddress,omitempty"`
+	PreferredLanguage *string `json:"preferredLanguage,omitempty"`
+	Locale            *string `json:"locale,omitempty"`
+	Timezone          *string `json:"timezone,omitempty"`
+	UserType          *string `json:"userType,omitempty"`
+	EmployeeNumber    *string `json:"employeeNumber,omitempty"`
+	CostCenter        *string `json:"costCenter,omitempty"`
+	Organization      *string `json:"organization,omitempty"`
+	Division          *string `json:"division,omitempty"`
+	Department        *string `json:"department,omitempty"`
+	ManagerId         *string `json:"managerId,omitempty"`
+	Manager           *string `json:"manager,omitempty"`
+}
+
+// Credentials is a redacted Okta user's credential details. Only the credential provider is retained.
+//
+// See https://developer.okta.com/docs/reference/api/users/#credentials-object for details.
+type Credentials struct {
+	Password         *struct{} `json:"password,omitempty"`          // Contains "value"; omit but mark.
+	RecoveryQuestion *struct{} `json:"recovery_question,omitempty"` // Contains "question" and "answer"; omit but mark.
+	Provider         Provider  `json:"provider"`
+}
+
+// Provider is an Okta credential provider.
+//
+// See https://developer.okta.com/docs/reference/api/users/#provider-object for details.
+type Provider struct {
+	Type string  `json:"type"`
+	Name *string `json:"name,omitempty"`
+}
+
+// Response is a set of omit options specifying a part of the response to omit.
+//
+// See https://developer.okta.com/docs/reference/api/users/#content-type-header-fields-2 for details.
+type Response uint8
+
+const (
+	// Omit the credentials sub-object from the response.
```

Comment:
My editor flags these docstrings as not beginning with the variable name (`// OmitCredentials omits ...`. I'm not sure if this is something we enforce (the linter doesn't seem to care), but thought I'd mention it.

Reply:
I'm OK with this break in style. The intention is clear.